### PR TITLE
logger: remove prints and debug flag

### PR DESCRIPTION
**Pull request by geyslan**
## Description (git log)

0d74743e docs: new flag updates
32ebdbd9 pkg/logger: fix key/value fields order
62f6a5e5 pkg/cmd/urfave: join osinfo logs
01232f7e logger: replace prints with logger
4aa037f8 logger: replace debug with log flag

Fixes: #2368
Fixes: #2369
Fixes: #2484

## Type of change

- [x] Bug fix (non-breaking change fixing an issue, preferable).
- [x] Code refactor (code improvement and/or code removal)

### CODE REVIEW

**Comment by geyslan** [comment]
@NDStrahilevitz This PR removes the debug flag, but I didn't touch the documentation since I remembered this discussion: https://github.com/aquasecurity/tracee/pull/2403#discussion_r1037150342

Are you still decided to remove the env vars logic? If so, I must bring the debug flag back. WDYT?

**Comment by AlonZivony** [comment]
> > @NDStrahilevitz This PR removes the debug flag, but I didn't touch the documentation since I remembered this discussion: [#2403 (comment)](https://github.com/aquasecurity/tracee/pull/2403#discussion_r1037150342)
> > Are you still decided to remove the env vars logic? If so, I must bring the debug flag back. WDYT?
> 
> We should remove the env vars AND add an `InitLogger` method so outside consumers can programatically set the logger themselves. And yes I guess we would need a debug flag but it could be much simpler than the debug flag we had before with the `Enable` `Disable` and all that, it would just be a bool that we use to `InitLogger` or `SetLevel` globally at the start something like that. Sounds reasonable?

I agree with @NDStrahilevitz here - we should still have a `debug` flag, which change log level to be `debug` or lower.
This way the command line API will enable some control over the logger.
On the other hand, you can have a flag determining the level of the logger for it to be even more configurable.

**Comment by yanivagman** [comment]
> On the other hand, you can have a flag determining the level of the logger for it to be even more configurable.

I would go for that. Something like `--log-level=debug` (and other log levels possible)

**Comment by geyslan** [comment]
> > On the other hand, you can have a flag determining the level of the logger for it to be even more configurable.
> 
> I would go for that. Something like `--log debug` (and other log levels possible)

Alright, `--log debug` for the win.

File: docs/contributing/building/nocore-ebpf.md [hunk 1]
```diff
@@ -118,36 +118,38 @@ If you install the non CO-RE eBPF object and run **tracee-ebpf** in an
...
+{"level":"debug","ts":1670617146.8852425,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"VERSION_CODENAME","OS_KERNEL_RELEASE":"focal"}
+{"level":"debug","ts":1670617146.885269,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"KERNEL_RELEASE","OS_KERNEL_RELEASE":"5.4.0-132-generic"}
+{"level":"debug","ts":1670617146.8852894,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ARCH","OS_KERNEL_RELEASE":"x86_64"}
+{"level":"debug","ts":1670617146.8853118,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"VERSION","OS_KERNEL_RELEASE":"\"20.04.5 LTS (Focal Fossa)\""}
+{"level":"debug","ts":1670617146.8853261,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID","OS_KERNEL_RELEASE":"ubuntu"}
+{"level":"debug","ts":1670617146.8853502,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID_LIKE","OS_KERNEL_RELEASE":"debian"}
+{"level":"debug","ts":1670617146.885369,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"PRETTY_NAME","OS_KERNEL_RELEASE":"\"Ubuntu 20.04.5 LTS\""}
```
Comment (yanivagman) [inline docs/contributing/building/nocore-ebpf.md:129]: Can you please remind me why do we have the extra OS_KERNEL_RELEASE in all of the above messages?

File: docs/contributing/building/nocore-ebpf.md [hunk 2]
```diff
@@ -118,36 +118,38 @@ If you install the non CO-RE eBPF object and run **tracee-ebpf** in an
...
+{"level":"debug","ts":1670617146.8881311,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":101,"security_lockdown":"none"}
+{"level":"debug","ts":1670617146.8965306,"msg":"BTF","pkg":"initialize","file":"bpfobject.go","line":40,"bpfenv":false,"btfenv":false,"vmlinux":true}
```
Comment (yanivagman) [inline docs/contributing/building/nocore-ebpf.md:131]: This doesn't correspond the environment described above where vmlinux should be false as well (thus use noncore)

File: docs/contributing/building/nocore-ebpf.md [hunk 3]
```diff
@@ -118,36 +118,38 @@ If you install the non CO-RE eBPF object and run **tracee-ebpf** in an
...
+{"level":"debug","ts":1670617459.811008,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"KERNEL_RELEASE","OS_KERNEL_RELEASE":"5.4.0-132-generic"}
+{"level":"debug","ts":1670617459.81103,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ARCH","OS_KERNEL_RELEASE":"x86_64"}
+{"level":"debug","ts":1670617459.8110468,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"VERSION","OS_KERNEL_RELEASE":"\"20.04.5 LTS (Focal Fossa)\""}
+{"level":"debug","ts":1670617459.8110704,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID","OS_KERNEL_RELEASE":"ubuntu"}
+{"level":"debug","ts":1670617459.81109,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID_LIKE","OS_KERNEL_RELEASE":"debian"}
+{"level":"debug","ts":1670617459.8134959,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":101,"security_lockdown":"none"}
+{"level":"debug","ts":1670617459.818108,"msg":"BTF","pkg":"initialize","file":"bpfobject.go","line":40,"bpfenv":true,"btfenv":false,"vmlinux":true}
```
Comment (yanivagman) [inline docs/contributing/building/nocore-ebpf.md:151]: same here - environments don't correspond (and debug messages are different)

File: docs/contributing/building/nocore-ebpf.md [hunk 4]
```diff
@@ -174,21 +176,22 @@ tracee@f65bab137305[/tracee]$ make clean
...
+{"level":"debug","ts":1670617890.4573712,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"VERSION","OS_KERNEL_RELEASE":"\"20.04.5 LTS (Focal Fossa)\""}
+{"level":"debug","ts":1670617890.4573927,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID","OS_KERNEL_RELEASE":"ubuntu"}
+{"level":"debug","ts":1670617890.4574254,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":37,"OSReleaseField":"ID_LIKE","OS_KERNEL_RELEASE":"debian"}
+{"level":"debug","ts":1670617890.4582794,"msg":"osinfo","pkg":"urfave","file":"urfave.go","line":101,"security_lockdown":"none"}
+{"level":"warn","ts":1670617890.4583886,"msg":"KConfig: could not check enabled kconfig features","error":"could not read /boot/config-5.4.0-132-generic: stat /boot/config-5.4.0-132-generic: no such file or directory"}
+{"level":"warn","ts":1670617890.4584002,"msg":"KConfig: assuming kconfig values, might have unexpected behavior"}
+{"level":"debug","ts":1670617890.458432,"msg":"BTF","pkg":"initialize","file":"bpfobject.go","line":40,"bpfenv":false,"btfenv":false,"vmlinux":true}
```
Comment (yanivagman) [inline docs/contributing/building/nocore-ebpf.md:191]: same

File: docs/contributing/building/nocore-ebpf.md (hunk 1 above)
Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md:129]: This is strange. I'll check it.

Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md:129]: This is a snapshot from a different environment; fixing. Thanks.

File: docs/contributing/building/nocore-ebpf.md (hunk 2 above)
Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md:131]: ditto. tks.

File: docs/contributing/building/nocore-ebpf.md (hunk 3 above)
Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md:151]: ditto. tks.

File: docs/contributing/building/nocore-ebpf.md (hunk 1 above)
Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md:129]: `OS_KERNEL_RELEASE` was being used as a log field. I'm making it clearer.

```log
{"level":"debug","ts":1670972052.3996286,"msg":"osinfo","VERSION_CODENAME":"focal","KERNEL_RELEASE":"5.4.0-91-generic","ARCH":"x86_64","VERSION":"\"20.04.5 LTS (Focal Fossa)\"","ID":"ubuntu","ID_LIKE":"debian","PRETTY_NAME":"\"Ubuntu 20.04.5 LTS\"","VERSION_ID":"\"20.04\"","pkg":"urfave","file":"urfave.go","line":53}
```

Comment (geyslan) [reply docs/contributing/building/nocore-ebpf.md]: Done.

**Review by yanivagman — APPROVED** [review]
LGTM,
nice work @geyslan !

**Pull request merged**